### PR TITLE
Adding support for DotNetCoreSdkSearch …

### PR DESCRIPTION
**Pull request by powercode**
Issue:  wixtoolset/issues#7058
Summary of Changes:

- Adding support for SDK version detection in `netcoresearch.exe` using the target name "sdk"
- Adding a function `NetfxPerformDetectNetCoreSdk` to NetFX/be (lots of code duplication with `NetfxPerformDetectNetCore`, but opting for as little change as possible to existing code.
- Hooking up `DotNetCoreSdkSearch`  in `NetFxCompiler`. Extracting helper parsing functions for the parts that are common between the runtime and the SDK attributes.
- Adding symbol definition for `NetFxNetCoreSdkSearchSymbol` mimicking `NetFxNetCoreSearchSymbol` but without the `Runtime` field.

- Spelling of  `NetFxDotNetCompatibilityCheckSymbollFields` => `NetFxDotNetCompatibilityCheckSymbolFields` (double ll). This is a public enum - breaking change? Better to fix before release?

- Adding Test Fragments for `netfx:DotNetCoreSdkSearch` to `TestBAWixLib/NetCore609_*.wxs` and `UsingDotNetCorePackages/NetCore3.1.12_*.wxs`

### CODE REVIEW

**Comment by powercode** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by rseanhall — COMMENTED** [review]
Thanks for taking a stab at this. Unfortunately, the SDK versioning is more complex because of feature bands - https://learn.microsoft.com/en-us/dotnet/core/releases-and-support#feature-bands-sdk-only. I think we need the user to not only specify the target major version but also the target feature band. As their example shows, it's possible the bundle needs to install the SDK version 5.0.101 even if 5.0.200 is already installed.

The DotNetCoreSdkCompatibilityCheck functionality isn't finished in this PR. The Custom Action C++ code is missing here so it would fail at runtime. And even if was there, I don't think the Microsoft provided NetCoreCheck.exe supports detecting SDKs.

File: src/ext/NetFx/be/NetfxBundleExtension.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@ class CWixNetfxBundleExtension : public CBextBaseBundleExtension
     {
         HRESULT hr = S_OK;
 
-        hr = NetfxSearchExecute(&m_searches, wzId, wzVariable, m_pEngine, m_sczBaseDirectory);
+        hr = NetfxSearchExecute(&m_runtimeSearches, &m_sdkSearches, wzId, wzVariable, m_pEngine, m_sczBaseDirectory);
```
[the remark this answers is not in the document]
Comment (powercode) [reply src/ext/NetFx/be/NetfxBundleExtension.cpp:16]: Maybe renamed to m_searches?

**Comment by robmen** [comment]
@powercode WiX v4-rc.2 is releasing Jan 20th. Ideally, this feature will be completed before that date, or we'll move it to the next version.

**Comment by powercode** [comment]
@robmen Rewritten to fit better into the existing structure. Also added matching for feature bands.

**Comment by rseanhall** [comment]
Since it seemed like you thought we should keep the functionality of searching regardless of feature bands, I reverted netfx:DotNetCoreSdkSearch to only look at the major version like netfx:DotNetCoreSearch does. I added netfx:DotNetCoreSdkFeatureBandSearch for people that are trying to install an SDK in a bundle along with some cleanup.

**Comment by rseanhall** [comment]
Thanks!

**Pull request merged**